File: pset6/cash.py
# List of coin denominations in cents
arr = [25, 10, 5, 1]
ans = []

# Prompt the user for the change owed
while True:
    try:
        n = float(input("Change owed: ")) * 100
        if n >= 0:
            n = int(round(n))
            break
    except ValueError:
        pass

# Calculate the minimum number of coins
for coin in arr:
    while n >= coin:
        n -= coin
        ans.append(coin)

print("Coins used:")
for coin in ans:
    print(coin, end=" ")
print()

print("The minimum number of coins is", len(ans))
print("The coins are:", " ".join(map(str, ans)))

File: pset1/cash.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int arr[4] = {1, 5, 10, 25};
    int ans[100];
    int m = sizeof(arr) / sizeof(arr[0]);

    // Prompt the user for the pyramid's height
    int n;
    do
    {
        n = get_int("Change owed: ");
    }
    while (n < 0);

    int k = 0;
    for (int i = m - 1; i >= 0; i--)
    {
        while (n >= arr[i])
        {
            n -= arr[i];
            ans[k++] = arr[i];
        }
    }

    printf("Coins used:\n");
    for (int j = 0; j < k; j++)
    {
        printf("%d ", ans[j]);
    }
    printf("\n");
    printf("The minimum number of coins is %d\n", k);
    printf("The coins are: ");
    for (int i = 0; i < k; i++)
    {
        printf("%d ", ans[i]);
    }
    printf("\n");
}